### PR TITLE
[Darwin] New XPC Service

### PR DESCRIPTION
**Pull request by kiel-apple**
lighter / more direct XPC wrappers for MTRDevice and MTRDeviceController

### CODE REVIEW

File: src/darwin/Framework/CHIP/MTRXPCService.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
+//
```
Comment (jtung-apple) [inline src/darwin/Framework/CHIP/MTRXPCService.h:1]: Nit: please remove this line, and the same for the other files

File: src/darwin/Framework/CHIPTests/TestHelpers/MTRTestServerAppRunner.m [hunk 2]
```diff
@@ -21,11 +21,13 @@
 
 static unsigned sAppRunnerIndex = 1;
 
+#if HAVE_NSTASK
 // kBasePort gets the discriminator added to it to figure out the port the app
 // should be using.  This ensures that apps with distinct discriminators use
 // distinct ports.
 static const uint16_t kMinDiscriminator = 1111;
 static const uint16_t kBasePort = 5542 - kMinDiscriminator;
+#endif // HAVE_NSTASK
```
[the remark this answers is not in the document]
Comment (kiel-apple) [reply src/darwin/Framework/CHIPTests/TestHelpers/MTRTestServerAppRunner.m:30]: #34970

File: src/darwin/Framework/CHIP/MTRDeviceControllerOverXPCParameters.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Matter/Matter.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MTRDeviceControllerOverXPCParameters : MTRDeviceControllerAbstractParameters
```
Comment (woody-apple) [inline src/darwin/Framework/CHIP/MTRDeviceControllerOverXPCParameters.h:21]: Let's come up with a better name for this, I wish I had a better suggestion, but Over is not great. Maybe MTRDeviceControllerXPCParameters?

Comment (kiel-apple) [reply src/darwin/Framework/CHIP/MTRDeviceControllerOverXPCParameters.h:21]: Yep, unattached to the current name so that's as good as any in my view.  Making the change.

File: src/darwin/Framework/CHIP/MTRDeviceControllerXPCParameters.h [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Matter/Matter.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+static NSString * MatterPluginWellKnownServiceName = @"com.apple.home.matter.framework.xpc";
```
Comment (anush-apple) [inline src/darwin/Framework/CHIP/MTRDeviceControllerXPCParameters.h:21]: Why is this here?

Comment (kiel-apple) [reply src/darwin/Framework/CHIP/MTRDeviceControllerXPCParameters.h]: removed

**Review by woody-apple — APPROVED** [review]
